engine/src/main/java/org/hibernate/search/engine/search/predicate/dsl/QueryStringPredicateOptionsStep.java

```diff
@@ -0,0 +1,202 @@
+	//	 * when determining whether a term matched a value.
+	//	 * For example, the distance between wined and wind is 1.
+	//	 * Valid values are non-negative integers or AUTO.
+	//	 * The default, AUTO, chooses a value based on the length of each term and is a good choice for most use cases.
+	//	 *
+	//	 * TODO: AUTO == null
+	//	 *
+	//	 * @param fuzziness
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S fuzziness(int fuzziness);
+
+
+	//	/**
+	//	 * The maximum number of terms to which the query can expand.
+	//	 * Fuzzy queries “expand to” a number of matching terms that are within the distance specified in fuzziness.
+	//	 * Then OpenSearch tries to match those terms. Default is 50.
+	//	 * @param fuzzyMaxExpansions
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S fuzzyMaxExpansions(int fuzzyMaxExpansions);
+
+	//	/**
+	//	 * Setting fuzzy_transpositions to true (default) adds swaps of adjacent characters to the insert, delete,
+	//	 * and substitute operations of the fuzziness option.
+	//	 * For example, the distance between wind and wnid is 1 if fuzzy_transpositions is true (swap “n” and “i”) and 2
+	//	 * if it is false (delete “n”, insert “n”). If fuzzy_transpositions is false,
+	//	 * rewind and wnid have the same distance (2) from wind,
+	//	 * despite the more human-centric opinion that wnid is an obvious typo.
+	//	 * The default is a good choice for most use cases.
+	//	 *
+	//	 * @param fuzzyTranspositions
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S fuzzyTranspositions(boolean fuzzyTranspositions);
+
+	//	/**
+	//	 * Setting lenient to true ignores data type mismatches between the query and the document field.
+	//	 * For example, a query string of "8.2" could match a field of type float. Default is false.
+	//	 *
+	//	 * @param lenient
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S lenient(boolean lenient);
+
+	//	/**
+	//	 *
+	//	 * @param maxDeterminizedStates
+	//	 * @return {@code this}, for method chaining.
+	//	 */
+	//	S maxDeterminizedStates(int maxDeterminizedStates);
+
+	/**
+	 *
+	 * @param matchingClausesNumber
+	 * @return {@code this}, for method chaining.
+	 */
+	default S minimumShouldMatchNumber(int matchingClausesNumber) {
+		return minimumShouldMatch()
+				.ifMoreThan( 0 ).thenRequireNumber( matchingClausesNumber )
+				.end();
+	}
+
+	/**
+	 *
+	 * @param matchingClausesPercent
+	 * @return {@code this}, for method chaining.
+	 */
+	default S minimumShouldMatchPercent(int matchingClausesPercent) {
+		return minimumShouldMatch()
+				.ifMoreThan( 0 ).thenRequirePercent( matchingClausesPercent )
+				.end();
+	}
+
+	MinimumShouldMatchConditionStep<? extends S> minimumShouldMatch();
+
+	/**
+	 *
+	 * @param constraintContributor
+	 * @return {@code this}, for method chaining.
+	 */
+	S minimumShouldMatch(
+			Consumer<? super MinimumShouldMatchConditionStep<?>> constraintContributor);
```

Comment:
OK, but please make sure the javadoc is consistent with what we already have elsewhere.

Also... Apparently it's supported for the simple query string query too: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-simple-query-string-query.html
So if you add it here, maybe open a Jira for simple query string? Even if you solve it in this PR: we need to document the improvement in the changelog.

Reply:
created https://hibernate.atlassian.net/browse/HSEARCH-5073 for tracking